File: VB/DXApplication5/Data/CustomAppointment.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq

Namespace DXApplication5
	Public Class CustomAppointment
		Private _Priority As Integer
		Private m_Start As DateTime
		Private m_End As DateTime
		Private m_Subject As String
		Private m_Status As Integer
		Private m_Description As String
		Private m_Label As Integer
		Private m_Location As String
		Private m_Allday As Boolean
		Private m_EventType As Integer
		Private m_RecurrenceInfo As String
		Private m_ReminderInfo As String
		Private m_OwnerId As Object


		Public Property StartTime() As DateTime
			Get
				Return m_Start
			End Get
			Set(ByVal value As DateTime)
				m_Start = value
			End Set
		End Property
		Public Property EndTime() As DateTime
			Get
				Return m_End
			End Get
			Set(ByVal value As DateTime)
				m_End = value
			End Set
		End Property
		Public Property Subject() As String
			Get
				Return m_Subject
			End Get
			Set(ByVal value As String)
				m_Subject = value
			End Set
		End Property
		Public Property Status() As Integer
			Get
				Return m_Status
			End Get
			Set(ByVal value As Integer)
				m_Status = value
			End Set
		End Property
		Public Property Description() As String
			Get
				Return m_Description
			End Get
			Set(ByVal value As String)
				m_Description = value
			End Set
		End Property
		Public Property Label() As Integer
			Get
				Return m_Label
			End Get
			Set(ByVal value As Integer)
				m_Label = value
			End Set
		End Property
		Public Property Location() As String
			Get
				Return m_Location
			End Get
			Set(ByVal value As String)
				m_Location = value
			End Set
		End Property
		Public Property AllDay() As Boolean
			Get
				Return m_Allday
			End Get
			Set(ByVal value As Boolean)
				m_Allday = value
			End Set
		End Property
		Public Property EventType() As Integer
			Get
				Return m_EventType
			End Get
			Set(ByVal value As Integer)
				m_EventType = value
			End Set
		End Property
		Public Property RecurrenceInfo() As String
			Get
				Return m_RecurrenceInfo
			End Get
			Set(ByVal value As String)
				m_RecurrenceInfo = value
			End Set
		End Property
		Public Property ReminderInfo() As String
			Get
				Return m_ReminderInfo
			End Get
			Set(ByVal value As String)
				m_ReminderInfo = value
			End Set
		End Property
		Public Property OwnerId() As Object
			Get
				Return m_OwnerId
			End Get
			Set(ByVal value As Object)
				m_OwnerId = value
			End Set
		End Property

		Public Property PriorityId() As Integer
			Get
				Return _Priority
			End Get
			Set(ByVal value As Integer)
				_Priority = value
			End Set
		End Property

		Public Sub New()
		End Sub


	End Class


	Public Enum PriorityStatus
		High = 0
		Medium
		Low
	End Enum
	Public Class Priority

		Public Sub New()

		End Sub
		Private _DisplayName As String
		Private _Type As PriorityStatus

		Public Property Type() As PriorityStatus
			Get
				Return _Type
			End Get
			Set(ByVal value As PriorityStatus)
				_Type = value
			End Set
		End Property


		Public Property DisplayName() As String
			Get
				Return _DisplayName
			End Get
			Set(ByVal value As String)
				_DisplayName = value
			End Set
		End Property



	End Class
End Namespace

File: CS/DXApplication5/Data/CustomAppointment.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;

namespace DXApplication5
{
    public class CustomAppointment
    {
        private int _Priority;
        private DateTime m_Start;
        private DateTime m_End;
        private string m_Subject;
        private int m_Status;
        private string m_Description;
        private int m_Label;
        private string m_Location;
        private bool m_Allday;
        private int m_EventType;
        private string m_RecurrenceInfo;
        private string m_ReminderInfo;
        private object m_OwnerId;
     

        public DateTime StartTime { get { return m_Start; } set { m_Start = value; } }
        public DateTime EndTime { get { return m_End; } set { m_End = value; } }
        public string Subject { get { return m_Subject; } set { m_Subject = value; } }
        public int Status { get { return m_Status; } set { m_Status = value; } }
        public string Description { get { return m_Description; } set { m_Description = value; } }
        public int Label { get { return m_Label; } set { m_Label = value; } }
        public string Location { get { return m_Location; } set { m_Location = value; } }
        public bool AllDay { get { return m_Allday; } set { m_Allday = value; } }
        public int EventType { get { return m_EventType; } set { m_EventType = value; } }
        public string RecurrenceInfo { get { return m_RecurrenceInfo; } set { m_RecurrenceInfo = value; } }
        public string ReminderInfo { get { return m_ReminderInfo; } set { m_ReminderInfo = value; } }
        public object OwnerId { get { return m_OwnerId; } set { m_OwnerId = value; } }

        public int PriorityId
        {
            get
            {
                return _Priority;
            }
            set
            {
                _Priority = value;
            }
        }

        public CustomAppointment()
        {
        }


    }


    public enum PriorityStatus { High = 0, Medium , Low }
    public class Priority
    {
        
        public Priority()
        {
            
        }
        private string _DisplayName;
        private PriorityStatus _Type;

        public PriorityStatus Type
        {
            get { return _Type; }
            set
            {
                _Type = value;
            }
        }


        public string DisplayName
        {
            get { return _DisplayName; }
            set
            {
                _DisplayName = value;
            }
        }
        
        
        
    }
}